File: libs/shared-types/python-src/debrief/types/states/current_state.py
"""CurrentState Pydantic model for maritime analysis complete state."""

from pydantic import BaseModel, Field
from .editor_state import EditorState


class CurrentState(BaseModel):
    """Complete current state information for a Debrief editor including metadata."""

    editorId: str = Field(
        ...,
        description="Unique identifier for the editor"
    )
    filename: str = Field(
        ...,
        description="Filename of the document being edited"
    )
    editorState: EditorState = Field(
        ...,
        description="Complete editor state containing all sub-states"
    )
    historyCount: int = Field(
        ...,
        ge=0,
        description="History count for this editor"
    )

    model_config = {
        "extra": "forbid",
        "json_schema_extra": {
            "$id": "https://schemas.debrief.com/states/current-state.schema.json"
        }
    }

File: libs/shared-types/src/zod/states/current_state.ts
import { z } from "zod"

export default z.object({ "editorId": z.string().describe("Unique identifier for the editor"), "filename": z.string().describe("Filename of the document being edited"), "editorState": z.any().describe("Complete editor state containing all sub-states"), "historyCount": z.number().int().gte(0).describe("History count for this editor") }).strict().describe("Complete current state information for a Debrief editor including metadata.")
